rust/actyx/certs/src/app_manifest.rs

```diff
@@ -103,65 +67,34 @@ impl AppManifest {
         };
         let dev_signature = Signature::new(&hash_input, dev_privkey)?;
         let manifest_signature = AppManifestSignature::new(dev_signature, dev_cert);
-        Ok(Self(AppManifestIo {
-            app_id,
-            display_name,
-            version,
-            signature: Some(manifest_signature),
-        }))
-    }
-
-    pub fn signed(app_id: AppId, display_name: String, version: String, signature: AppManifestSignature) -> Self {
-        Self(AppManifestIo {
+        let manifest_signature_string: String = manifest_signature.try_into()?;
+        let manifest: AppManifest = AppManifestIo {
```

Comment:
Why use `AppManifestIo` here instead of just creating an `AppManifest`? That would also make the `internal` feature unnecessary.

Reply:
If we remove AppManifestIo, does the `TryInto` stuff go to the deserializer?

Reply:
Not sure I understand the question: what I mean is that `AppManifestIo` is (and shall remain) a purely internal detail of how `AppManifest` is (de)serialized. So just remove the export and the `internals` feature and directly construct an `AppManifest` here.

Reply:
Ah, I see.
Currently, `signed` manifest is purely created in the Actyx (by constructing `AppManifestIo` manually), no API is available to the user for creating a `signed` manifest. That's why the `internals` bridge.